modules/notifications/notifications.php

```diff
@@ -727,69 +727,98 @@ function schedule_emails( $recipients, $subject, $message, $message_headers = ''
-	 * @return string or array of recipients to receive notification 
+	 * @param WP_Post $post
+	 * @param bool $string Whether to return recipients as comma-delimited string or array.
+	 * @return string|array Recipients to receive notification.
 	 */
 	private function _get_notification_recipients( $post, $string = false ) {
 		global $edit_flow;
-		
+
 		$post_id = $post->ID;
-		if( !$post_id ) return;
-		
-		$authors = array();
-		$admins = array();
-		$recipients = array();
+		if ( ! $post_id ) {
+			return;
+		}
 
-		// Email all admins, if enabled
-		if( 'on' == $this->module->options->always_notify_admin )
+		// Email all admins if enabled.
+		$admins = array();
+		if ( 'on' === $this->module->options->always_notify_admin ) {
 			$admins[] = get_option('admin_email');
-		
+		}
+
 		$usergroup_users = array();
 		if ( $this->module_enabled( 'user_groups' ) ) {
-			// Get following users and usergroups
+			// Get following users and usergroups.
 			$usergroups = $this->get_following_usergroups( $post_id, 'ids' );
-			foreach( (array)$usergroups as $usergroup_id ) {
+			foreach ( (array) $usergroups as $usergroup_id ) {
 				$usergroup = $edit_flow->user_groups->get_usergroup_by( 'id', $usergroup_id );
-				foreach( (array)$usergroup->user_ids as $user_id ) {
+				foreach ( (array) $usergroup->user_ids as $user_id ) {
 					$usergroup_user = get_user_by( 'id', $user_id );
-					if ( $usergroup_user && is_user_member_of_blog( $user_id ) )
+					if ( $this->user_can_be_notified( $usergroup_user, $post_id ) ) {
 						$usergroup_users[] = $usergroup_user->user_email;
+					}
 				}
 			}
 		}
-		
+
 		$users = $this->get_following_users( $post_id, 'user_email' );
-		
-		// Merge arrays and filter any duplicates
-		$recipients = array_merge( $authors, $admins, $users, $usergroup_users );
-		$recipients = array_unique( $recipients );
-
-		// Process the recipients for this email to be sent
-		foreach( $recipients as $key => $user_email ) {
-			// Get rid of empty email entries
-			if ( empty( $recipients[$key] ) )
-				unset( $recipients[$key] );
-			// Don't send the email to the current user unless we've explicitly indicated they should receive it
-			if ( false === apply_filters( 'ef_notification_email_current_user', false ) && wp_get_current_user()->user_email == $user_email )
-				unset( $recipients[$key] );
+		foreach( (array) $users as $key => $user ) {
+			$user_object = get_user_by( 'email', $user );
+			if ( ! $this->user_can_be_notified( $user_object, $post_id ) ) {
+				unset( $users[ $key ] );
+			}
 		}
-		
-		// Filter to allow further modification of recipients
+
+		// Merge arrays and filter any duplicates.
+		$recipients = array_unique( array_merge( $admins, $users, $usergroup_users ) );
+
+		// Process the recipients for this email to be sent.
+		foreach(  $recipients as $key => $user_email ) {
+			// Get rid of empty email entries.
+			if ( empty( $recipients[ $key ] ) ) {
```

Comment:
Hmm, good question. This check was legacy code that I left there, but re-reading looks like a possible typo. Should be checking if the email is empty, because it is technically possible to have a user without an email address.

Instead of the conditional, could just wrap `array_filter()` around the array manipulations above which will simply strip array any empty values when used without a callback function.